app/javascript/pages/listings/listing-detail.tsx

```diff
@@ -74,10 +74,12 @@ const ListingDetail = () => {
 
   useEffect(() => {
     const path = getPathWithoutLanguagePrefix(router.pathname)
-    void getListing(path.split("/")[2]).then((listing: RailsListing) => {
+    const previewFlag = router.asPath.includes("preview=true")
```

Comment:
If you were looking to make this a little cleaner, you could leverage [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get#examples) in our `Router.ts` file:

```js
// ...
get asPath(): string {
  return window.location.pathname + window.location.search
},
get search(queryParam: string): string | null {
  return new URLSearchParams(window.location.search).get(queryParam)
}
// ...
```

Although this is more code, and I'm not sure if it's worth implementing.